source/common/secret/sds_api.cc

```diff
@@ -143,13 +143,24 @@ absl::Status SdsApi::onConfigUpdate(const std::vector<Config::DecodedResourceRef
   return absl::OkStatus();
 }
 
-absl::Status SdsApi::onConfigUpdate(const std::vector<Config::DecodedResourceRef>& added_resources,
-                                    const Protobuf::RepeatedPtrField<std::string>&,
-                                    const std::string&) {
-  const absl::Status status = validateUpdateSize(added_resources.size());
+absl::Status
+SdsApi::onConfigUpdate(const std::vector<Config::DecodedResourceRef>& added_resources,
+                       const Protobuf::RepeatedPtrField<std::string>& removed_resources,
+                       const std::string&) {
+  const absl::Status status = validateUpdateSize(added_resources.size(), removed_resources.size());
   if (!status.ok()) {
     return status;
   }
+
+  if (removed_resources.size() == 1) {
+    // We only removed a resource here; don't go through the SotW init process
+    // and expect this to be cleaned up by the owning cluster or listener.
```

Comment:
Please expand on the comment here. I think the key idea is this is a singleton resource subscription, so in normal operation, an xDS server should not be removing the resource until the referencing cluster / listener is no longer using it. 

@adisuissa why don't we structurally forbid singleton resource subscriptions from having removals in the baseline xDS infra, wondering why something like SDS needs to worry about this?

Reply:
+1 to enforcing this in the generic xDS machinery. I also think more documentation around "singleton" resources would be helpful as well; in any case, I've expanded the comment